rbac/internal/views.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for internal tenant management."""
+import datetime
+import json
+import logging
+
+import pytz
+from django.conf import settings
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from management.models import Group, Role
+from tenant_schemas.utils import tenant_context
+
+from api.models import Tenant
+
+
+logger = logging.getLogger(__name__)
+
+
+def destructive_ok():
+    """Determine if it's ok to run destructive operations."""
+    now = datetime.datetime.utcnow().replace(tzinfo=pytz.UTC)
+    return now < settings.INTERNAL_DESTRUCTIVE_API_OK_UNTIL
+
+
+def tenant_is_unmodified():
+    """Determine whether or not the tenant has been modified."""
+    group_qs = Group.objects.filter(system=True)
+    role_qs = Role.objects.filter(system=True)
+    only_system_group = group_qs and Group.objects.count() == 1
+    only_system_roles = role_qs and Role.objects.count() == role_qs.count()
+    return only_system_group and only_system_roles
+
+
+def list_unmodified_tenants(request):
+    """List unmodified tenants."""
+    logger.info(f"Unmodified tenants requested by: {request.user.username}")
+    tenant_qs = Tenant.objects.all()
+    to_return = []
+    for tenant_obj in tenant_qs:
```

Comment:
I am wondering how long will this list take in prod. There is no progress bar for it so maybe just wait 😁

Reply:
I just updated a few of these queries to just return the counts, so hopefully that will also help.